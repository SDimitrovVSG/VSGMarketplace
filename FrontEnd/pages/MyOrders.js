import { navbar } from "../components/navbar.js";
import { myOrdersApp } from "../src/myOrdersApp.js";

const MyOrders = () => {
    document.querySelector('#root').innerHTML = `
    ${navbar}
    <main id="myOrdersMain">
    <div id="headingSection">
        <span class="nameColumn">Name</span>
        <span class="qtyColumn">QTY</span>
        <span class="priceColumn">Price</span>
        <span class="orderDateColumn">Order Date</span>
        <span>Status</span>
    </div>
    </main>
    `;
    myOrdersApp();
};

export default MyOrders;
