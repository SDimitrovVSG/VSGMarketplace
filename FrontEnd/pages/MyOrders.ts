import { myOrderComponent } from "../components/myOrderComponent.js";
import { loadMyOrders } from "../src/itemsService.ts";
import { IOrder } from "../src/types.ts";

const MyOrders = async (): Promise<void> => {
    const main = document.querySelector("main") as HTMLElement;
    main.id = "myOrdersMain";
    main.innerHTML = `
    <div id="headingSection">
        <span class="nameColumn">Name</span>
        <span class="qtyColumn">QTY</span>
        <span class="priceColumn">Price</span>
        <span class="orderDateColumn">Order Date</span>
        <span class="orderStatus">Status</span>
    </div>
    `;

    const user: string | null = sessionStorage.getItem("user");
    const email: string = user ? JSON.parse(user).username : "eredzhepov@vsgbg.com";

    const orders = await loadMyOrders(email);
    orders.forEach((p: IOrder) => {
        myOrderComponent(p);
    });
};

export default MyOrders;
