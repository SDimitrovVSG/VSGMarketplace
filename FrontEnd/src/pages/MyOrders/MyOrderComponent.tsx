import { useRejectOrderMutation } from '../../services/ordersService.ts';
import { useState } from 'react';
import { IOrder } from '../../types/types.ts';
import { toast } from 'react-toastify';
import { Fade } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import Popup from '../../components/Popup.tsx';

type MyOrderProps = {
    order: IOrder;
}

const MyOrderComponent = ({ order }: MyOrderProps) => {
    const [rejectOrder] = useRejectOrderMutation();
    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
    const [currentStatus, setCurrentStatus] = useState(order.status);

    const popupMessage = `Are you sure you want to reject this order ?`;

    const onCancel = async () => {
        const response = await rejectOrder(order.id);
        setAnchorEl(null);
        if ('error' in response) {
            return;
        }

        setTimeout(() => {
            setCurrentStatus('Declined');
        }, 600);

        toast.info('Order declined.');
    }

    return (
        <>
            <Popup popupMessage={popupMessage} onYes={onCancel} anchorEl={anchorEl} setAnchorEl={setAnchorEl} />
            <Fade in={true} timeout={1000}>
                <div id={order.id.toString()} className='order' role='cell'>
                    <span className='nameColumn'>{order.productFullName}</span>
                    <div className='firstTwo'>
                        <span className='qtyColumn'>{order.quantity}</span>
                        <span className='priceColumn'>{order.price} BGN</span>
                    </div>
                    <span className='orderDateColumn'>{order.date}</span>
                    <div className='orderStatus'>
                        <span>{currentStatus}</span>
                        {currentStatus === 'Pending' &&
                            <a className='cancelOrder' onClick={e => setAnchorEl(e.currentTarget)}>
                                <CloseIcon />
                            </a>}
                    </div>
                </div>
            </Fade>
        </>
    );
};

export default MyOrderComponent;