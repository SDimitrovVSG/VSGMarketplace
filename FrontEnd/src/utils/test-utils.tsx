import { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import store from '../redux/store';
import { BrowserRouter } from 'react-router-dom';

type ProvidersProps = {
  children: ReactElement;
};

const AllTheProviders = ({ children }: ProvidersProps) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  )
}

const customRender = (ui: ReactElement, options?: object) => (
  render(ui, { wrapper: AllTheProviders, ...options })
);

export * from '@testing-library/react';

// override render method
export { customRender as render }