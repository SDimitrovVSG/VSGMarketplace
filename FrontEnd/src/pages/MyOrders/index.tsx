import { useGetMyOrdersQuery } from '../../services/ordersService.ts';
import { IOrder } from '../../types/types.ts';
import MyOrderComponent from './MyOrderComponent.tsx';

const MyOrders = () => {
    const { data: orders, isLoading } = useGetMyOrdersQuery();

    return (
        <main id='myOrdersMain'>
            <div id='headingSection'>
                <span className='nameColumn'>Name</span>
                <span className='qtyColumn'>QTY</span>
                <span className='priceColumn'>Price</span>
                <span className='orderDateColumn'>Order Date</span>
                <span className='orderStatus'>Status</span>
            </div>
            {orders?.length !== 0 && (
                orders?.map((order: IOrder) => (
                    <MyOrderComponent order={order} key={order.id} />
                ))
            )}
            {(orders?.length === 0 || !orders) && !isLoading && <div className='order'>No Orders</div>}
            {isLoading && <div className='order'>Loading...</div>}
        </main>
    );
};

export default MyOrders;
