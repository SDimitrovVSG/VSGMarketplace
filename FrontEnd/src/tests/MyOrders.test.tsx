import { render, screen } from 'test-utils';
import mockServer from 'mock-server';
import MyOrders from '../pages/MyOrders';

describe('MyOrders', () => {
    beforeAll(() => mockServer.listen());

    afterEach(() => mockServer.resetHandlers());

    afterAll(() => mockServer.close());
    it('should render with orders', async () => {
        render(<MyOrders />);
        const orders = await screen.findAllByRole('cell');
        expect(orders).toHaveLength(2);
    });
    it('should match the snapshot', async () => {
        const { container } = render(<MyOrders />);
        expect(container).toMatchSnapshot();
    });
});