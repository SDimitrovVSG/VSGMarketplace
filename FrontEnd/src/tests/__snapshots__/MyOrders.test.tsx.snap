// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOrders should match the snapshot 1`] = `
<div>
  <main
    id="myOrdersMain"
  >
    <div
      id="headingSection"
    >
      <span
        class="nameColumn"
      >
        Name
      </span>
      <span
        class="qtyColumn"
      >
        QTY
      </span>
      <span
        class="priceColumn"
      >
        Price
      </span>
      <span
        class="orderDateColumn"
      >
        Order Date
      </span>
      <span
        class="orderStatus"
      >
        Status
      </span>
    </div>
    <div
      class="order"
      role="cell"
      style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <span
        class="nameColumn"
      >
        1
      </span>
      <div
        class="firstTwo"
      >
        <span
          class="qtyColumn"
        >
          1
        </span>
        <span
          class="priceColumn"
        >
          12
           BGN
        </span>
      </div>
      <span
        class="orderDateColumn"
      >
        2023-05-31 04:16
      </span>
      <div
        class="orderStatus"
      >
        <span>
          Declined
        </span>
        <a
          class="cancelOrder"
        />
      </div>
    </div>
    <div
      class="order"
      role="cell"
      style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <span
        class="nameColumn"
      >
        2
      </span>
      <div
        class="firstTwo"
      >
        <span
          class="qtyColumn"
        >
          2
        </span>
        <span
          class="priceColumn"
        >
          12
           BGN
        </span>
      </div>
      <span
        class="orderDateColumn"
      >
        2023-05-31 04:17
      </span>
      <div
        class="orderStatus"
      >
        <span>
          Pending
        </span>
        <a
          class="cancelOrder"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </a>
      </div>
    </div>
  </main>
</div>
`;
